core/src/jsMain/kotlin/Advertisement.kt

```diff
@@ -1,12 +1,50 @@
 package com.juul.kable
 
+import com.benasher44.uuid.Uuid
+import com.juul.kable.external.BluetoothAdvertisingEvent
 import com.juul.kable.external.BluetoothDevice
+import com.juul.kable.external.JsIterator
+import com.juul.kable.external.iterable
+import org.khronos.webgl.DataView
 
 public actual class Advertisement internal constructor(
-    internal val bluetoothDevice: BluetoothDevice,
-    public actual val rssi: Int,
+    private val advertisement: BluetoothAdvertisingEvent,
 ) {
+    internal val bluetoothDevice: BluetoothDevice
+        get() = advertisement.device
 
     public actual val name: String?
         get() = bluetoothDevice.name
+
+    public actual val rssi: Int
+        get() = advertisement.rssi ?: 0
```

Comment:
It's bizarre that JS has rssi as nullable. When would it ever not know RSSI for a received advertisement?! 🤷 

Not really liking falling back to `0`, as that is a valid rssi.
Perhaps we should fallback to `Interger.MIN_VALUE`)? Which is well outside the valid range (so could be conditionally checked for).

Reply:
Good catch. With a 32bit integer, using `Int.MIN_VALUE` as the RSSI a distance calculation would give a result of the order of 10^10^9 meters, which is essentially infinite. Or NaN if the overflow isn't dealt with 😄